src/ploigos_step_runner/step_implementers/automated_governance/rekor.py

```diff
@@ -0,0 +1,277 @@
+
+        """
+        return DEFAULT_CONFIG
+
+    @staticmethod
+    def _required_config_or_result_keys():
+        """Getter for step configuration or previous step result artifacts that are required before
+        running this step.
+
+        See Also
+        --------
+        _validate_required_config_or_previous_step_result_artifact_keys
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys or previous step result artifacts
+            that are required before running the step.
+        """
+        return REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS
+
+    @staticmethod
+    def base64_encode(
+            file_path
+    ):
+        """Given a file_path, read and encode the contents in base64
+        Returns
+        -------
+        Base64Contents
+            base64 encoded string of file contents
+        """
+        encoding = Path(file_path).read_text().encode('utf-8')
+        return base64.b64encode(encoding).decode('utf-8')
+
+    def get_file_hash(self, file_path):
+        """Returns file hash of given file.
+
+        Returns
+        -------
+        StepResult
+            Object containing the dictionary results of this step.
+        """
+        sha256_hash = hashlib.sha256()
+        with open(file_path, "rb") as f:
+            # Read and update hash string value in blocks of 4K
+            for byte_block in iter(lambda: f.read(4096), b""):
+                sha256_hash.update(byte_block)
+        return sha256_hash.hexdigest()
+
+    def create_rekor_entry( self,
+        public_key_path,
+        extra_data_file
+    ):
+        """Creates a rekor entry as a dict object.
+
+        Returns
+        -------
+        StepResult
+            Dict that contains rekor entry for upload with the cli
+        """
+        hash = self.get_file_hash(extra_data_file)
+        sig_file = extra_data_file + '.asc'
+        sig_file_path = Path(sig_file)
+        if sig_file_path.exists():
+            sig_file_path.unlink()
+        self.get_gpg_key(sig_file,extra_data_file)
+        base64_encoded_extra_data = self.base64_encode(extra_data_file)
+        rekor_entry = {
+            "kind": "rekord",
+            "apiVersion": "0.0.1",
+            "spec": {
+                "signature": {
+                    "format": "pgp",
+                    "content": self.base64_encode(sig_file),
+                    "publicKey": {
+                        "content": self.base64_encode(public_key_path)
+                    }
+                },
+                "data": {
+                    "content": base64_encoded_extra_data,
+                    "hash": {
+                        "algorithm": "sha256",
+                        "value": hash
+                    }
+                },
+                "extraData": base64_encoded_extra_data
+            }
+        }
+        return rekor_entry;
+
+    def get_gpg_key(self, sig_file, extra_data_file):
```

Comment:
this would probably be useful in other palces so this should move to a utils/pgp.py class for re-use

Reply:
https://github.com/ploigos/ploigos-step-runner/pull/146/commits/b4a40ad006b797e701e428f2a33abe4837753899